src/plugins/event-plugins/NavigationPlugin.ts

```diff
@@ -8,288 +8,99 @@ import {
-            domContentLoaded:
-                entryData.domContentLoadedEventEnd -
-                entryData.domContentLoadedEventStart,
-            domComplete: entryData.domComplete,
-            domProcessingTime: entryData.loadEventStart - entryData.responseEnd,
-            loadEventStart: entryData.loadEventStart,
-            loadEventTime: entryData.loadEventEnd - entryData.loadEventStart,
-
-            duration: entryData.duration,
-
-            headerSize: entryData.transferSize - entryData.encodedBodySize,
-            transferSize: entryData.transferSize,
-            compressionRatio:
-                entryData.encodedBodySize > 0
-                    ? entryData.decodedBodySize / entryData.encodedBodySize
-                    : 0,
-            navigationTimingLevel: 2
-        };
-
-        if (this.context?.record) {
-            this.context.record(
-                PERFORMANCE_NAVIGATION_EVENT_TYPE,
-                eventDataNavigationTimingLevel2
-            );
-        }
+            // Record
+            const e = entry as PerformanceNavigationTiming;
+            this.context?.record(PERFORMANCE_NAVIGATION_EVENT_TYPE, {
+                version: '2.0.0',
+                name: this.context.config.recordResourceUrl
+                    ? e.name
+                    : undefined,
+                entryType: NAVIGATION,
+                startTime: e.startTime,
+                duration: e.duration,
+                initiatorType: e.initiatorType,
+                nextHopProtocol: e.nextHopProtocol,
+                workerStart: e.workerStart,
+                redirectStart: e.redirectStart,
+                redirectEnd: e.redirectEnd,
+                fetchStart: e.fetchStart,
+                domainLookupStart: e.domainLookupStart,
+                domainLookupEnd: e.domainLookupEnd,
+                connectStart: e.connectStart,
+                connectEnd: e.connectEnd,
+                secureConnectionStart: e.secureConnectionStart,
+                requestStart: e.requestStart,
+                responseStart: e.responseStart,
+                responseEnd: e.responseEnd,
+                transferSize: e.transferSize,
+                encodedBodySize: e.encodedBodySize,
+                decodedBodySize: e.decodedBodySize,
+                domComplete: e.domComplete,
+                domContentLoadedEventEnd: e.domContentLoadedEventEnd,
+                domContentLoadedEventStart: e.domContentLoadedEventStart,
+                domInteractive: e.domInteractive,
+                loadEventEnd: e.loadEventEnd,
+                loadEventStart: e.loadEventStart,
+                redirectCount: e.redirectCount,
+                type: e.type,
+                unloadEventEnd: e.unloadEventEnd,
+                unloadEventStart: e.unloadEventStart
+            } as PerformanceNavigationTimingEvent);
+
+            // Teardown
+            this.po.disconnect();
```

Comment:
question: why do we have to disconnect here?

Reply:
There's only one navigation event, so there's no use for the observer after after the first entry.